From patchwork Wed Aug 28 07:59:04 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vitaly Kuznetsov <vkuznets@redhat.com>
X-Patchwork-Id: 11118137
Return-Path: <SRS0=KojS=WY=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3782B1800
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 28 Aug 2019 07:59:14 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 21541205C9
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 28 Aug 2019 07:59:14 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726566AbfH1H7M (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 28 Aug 2019 03:59:12 -0400
Received: from mx1.redhat.com ([209.132.183.28]:48746 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1726368AbfH1H7K (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 28 Aug 2019 03:59:10 -0400
Received: from mail-wr1-f72.google.com (mail-wr1-f72.google.com
 [209.85.221.72])
        (using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id 2A3F9C0568FA
        for <kvm@vger.kernel.org>; Wed, 28 Aug 2019 07:59:10 +0000 (UTC)
Received: by mail-wr1-f72.google.com with SMTP id k15so905166wrw.18
        for <kvm@vger.kernel.org>; Wed, 28 Aug 2019 00:59:10 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=lBboHqcZl+GeeqMr+xhVz73awkXrj2sOIO6d13g0iRE=;
        b=hwAgKVbVCk0yCwvqgeT2jo45OSCNg6Iuww63xF0YVrPMXdt32fNAZ6iI4IfRsrv6/y
         PYnSI+ZqZf9YgP2zA21x4YH/oFfWtbBaIWAlOChFAj7SbfkOXZQqMQTg2i2fme7IsCR9
         AHZ84iI2FCJOynlGpyF+UPEDni1j6CY4GJeHY19MOMxXSsA/1cvAlu0Q/JrZQGwhtXoL
         B474D15tvlFQ7n/Ly8jhgkTUQWrAcLAv+PT22Qf/mQXRaOwS+Es3mPTWtf6S3KZgWWMW
         rv26LMqjGgFS3ltGBXO2FEqvZtqAFyV7HBvDJ63i8UCz9IEpXkF0hZP//Lk+WBfnls0U
         B1fw==
X-Gm-Message-State: APjAAAVfvC2d9GTULm2ZpAXIKHEV8bNWfJxx2MjCugDZdbCDGTmydAMp
        VKdFPu60cujnujtX7IJwE5umTRVulis1xv2JTHIDodDsjrlIs1h6Tsud6TFsQfYII530O+qEtDa
        Dwo/d6cKQs2Pn
X-Received: by 2002:adf:f18c:: with SMTP id h12mr2713824wro.47.1566979148632;
        Wed, 28 Aug 2019 00:59:08 -0700 (PDT)
X-Google-Smtp-Source: 
 APXvYqyg75w3y/I6ECh2ht91PCD+cPI1HSKMT6eKRHsTiLiLoq2ckp9NA9BsenuangkEk7V6n9c+3A==
X-Received: by 2002:adf:f18c:: with SMTP id h12mr2713808wro.47.1566979148444;
        Wed, 28 Aug 2019 00:59:08 -0700 (PDT)
Received: from vitty.brq.redhat.com (nat-pool-brq-t.redhat.com.
 [213.175.37.10])
        by smtp.gmail.com with ESMTPSA id
 a190sm2448469wme.8.2019.08.28.00.59.07
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Aug 2019 00:59:07 -0700 (PDT)
From: Vitaly Kuznetsov <vkuznets@redhat.com>
To: kvm@vger.kernel.org
Cc: linux-kernel@vger.kernel.org, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>,
 Joerg Roedel <joro@8bytes.org>, Jim Mattson <jmattson@google.com>,
 Sean Christopherson <sean.j.christopherson@intel.com>,
 Roman Kagan <rkagan@virtuozzo.com>
Subject: [PATCH v2 1/2] KVM: x86: svm: remove unneeded nested_enable_evmcs()
 hook
Date: Wed, 28 Aug 2019 09:59:04 +0200
Message-Id: <20190828075905.24744-2-vkuznets@redhat.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190828075905.24744-1-vkuznets@redhat.com>
References: <20190828075905.24744-1-vkuznets@redhat.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Since commit 5158917c7b019 ("KVM: x86: nVMX: Allow nested_enable_evmcs to
be NULL") the code in x86.c is prepared to see nested_enable_evmcs being
NULL and in VMX case it actually is when nesting is disabled. Remove the
unneeded stub from SVM code.

Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Reviewed-by: Jim Mattson <jmattson@google.com>
---
 arch/x86/kvm/svm.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index 40175c42f116..6d52c65d625b 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -7122,13 +7122,6 @@ static int svm_unregister_enc_region(struct kvm *kvm,
 	return ret;
 }
 
-static int nested_enable_evmcs(struct kvm_vcpu *vcpu,
-				   uint16_t *vmcs_version)
-{
-	/* Intel-only feature */
-	return -ENODEV;
-}
-
 static bool svm_need_emulation_on_page_fault(struct kvm_vcpu *vcpu)
 {
 	unsigned long cr4 = kvm_read_cr4(vcpu);
@@ -7337,7 +7330,7 @@ static struct kvm_x86_ops svm_x86_ops __ro_after_init = {
 	.mem_enc_reg_region = svm_register_enc_region,
 	.mem_enc_unreg_region = svm_unregister_enc_region,
 
-	.nested_enable_evmcs = nested_enable_evmcs,
+	.nested_enable_evmcs = NULL,
 	.nested_get_evmcs_version = NULL,
 
 	.need_emulation_on_page_fault = svm_need_emulation_on_page_fault,

From patchwork Wed Aug 28 07:59:05 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vitaly Kuznetsov <vkuznets@redhat.com>
X-Patchwork-Id: 11118139
Return-Path: <SRS0=KojS=WY=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 0FB221395
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 28 Aug 2019 07:59:21 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id EF83E22DA7
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 28 Aug 2019 07:59:20 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726554AbfH1H7M (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 28 Aug 2019 03:59:12 -0400
Received: from mx1.redhat.com ([209.132.183.28]:33432 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1726457AbfH1H7M (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 28 Aug 2019 03:59:12 -0400
Received: from mail-wm1-f70.google.com (mail-wm1-f70.google.com
 [209.85.128.70])
        (using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id 71EE183F3D
        for <kvm@vger.kernel.org>; Wed, 28 Aug 2019 07:59:11 +0000 (UTC)
Received: by mail-wm1-f70.google.com with SMTP id d64so666694wmc.7
        for <kvm@vger.kernel.org>; Wed, 28 Aug 2019 00:59:11 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=HqAYNYe6Ug0GzLuBdmUlgIsXKA4zYv5rxY3t4CUPVFQ=;
        b=Z2fPBO6HB1cFzSu0VLAr0gm9zOHeLZku6DgAa1TQLDSGcrE0MA6p0J4dQf5d2aqHWV
         kcxC7daOX3yO4IiLY2ggFX9/DX7PKpUCgcM+itbcvGZA/3wjdmunjl7r+9gIsnTl+Mt4
         +mn/rncUmQxcSGxKaOajpZRw9WYZLPNyp0rVEVU9dnIxaG3OHRpOzOw1C982ZDYhTwbD
         Ibgx1sa6r+ekpvK+djJuWk2jC0Yppph5wR3vid6G4dyOjZYHlnz7wLQNjIY7U1RSC9A2
         SbIktNK+IJaR8GCFeghV8pWv6an8/4NpF5YhQEijcvBPtyhpq4aNIJu4AyA5TMdZJI/E
         8TkA==
X-Gm-Message-State: APjAAAUA9+4d/oIwIzKguOmhylEv9iwHImWSFcsH+hXWydG3nLQAEBDt
        gEch2KjvIMBMGkRBNK4GdgPI8NCKQ2wrZcIVx9gQrjhxhZ/jzVP+dgTSSElwsU+Oj9nzueUWrrD
        RQ6GUeAyqR67P
X-Received: by 2002:adf:82d4:: with SMTP id 78mr2668211wrc.85.1566979149801;
        Wed, 28 Aug 2019 00:59:09 -0700 (PDT)
X-Google-Smtp-Source: 
 APXvYqyAgGjUe2XPR9J6yRmKjEXklPepwazpcGdHUVGFoBlK2l13Y0XcluoZTglZdM8C09USuZ9JBQ==
X-Received: by 2002:adf:82d4:: with SMTP id 78mr2668192wrc.85.1566979149600;
        Wed, 28 Aug 2019 00:59:09 -0700 (PDT)
Received: from vitty.brq.redhat.com (nat-pool-brq-t.redhat.com.
 [213.175.37.10])
        by smtp.gmail.com with ESMTPSA id
 a190sm2448469wme.8.2019.08.28.00.59.08
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Aug 2019 00:59:08 -0700 (PDT)
From: Vitaly Kuznetsov <vkuznets@redhat.com>
To: kvm@vger.kernel.org
Cc: linux-kernel@vger.kernel.org, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>,
 Joerg Roedel <joro@8bytes.org>, Jim Mattson <jmattson@google.com>,
 Sean Christopherson <sean.j.christopherson@intel.com>,
 Roman Kagan <rkagan@virtuozzo.com>
Subject: [PATCH v2 2/2] KVM: x86: announce KVM_CAP_HYPERV_ENLIGHTENED_VMCS
 support only when it is available
Date: Wed, 28 Aug 2019 09:59:05 +0200
Message-Id: <20190828075905.24744-3-vkuznets@redhat.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190828075905.24744-1-vkuznets@redhat.com>
References: <20190828075905.24744-1-vkuznets@redhat.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

It was discovered that after commit 65efa61dc0d5 ("selftests: kvm: provide
common function to enable eVMCS") hyperv_cpuid selftest is failing on AMD.
The reason is that the commit changed _vcpu_ioctl() to vcpu_ioctl() in the
test and this one can't fail.

Instead of fixing the test is seems to make more sense to not announce
KVM_CAP_HYPERV_ENLIGHTENED_VMCS support if it is definitely missing
(on svm and in case kvm_intel.nested=0).

Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Reviewed-by: Jim Mattson <jmattson@google.com>
---
 arch/x86/kvm/x86.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index d1cd0fcff9e7..149511f47377 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -3106,7 +3106,6 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 	case KVM_CAP_HYPERV_EVENTFD:
 	case KVM_CAP_HYPERV_TLBFLUSH:
 	case KVM_CAP_HYPERV_SEND_IPI:
-	case KVM_CAP_HYPERV_ENLIGHTENED_VMCS:
 	case KVM_CAP_HYPERV_CPUID:
 	case KVM_CAP_PCI_SEGMENT:
 	case KVM_CAP_DEBUGREGS:
@@ -3183,6 +3182,9 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 		r = kvm_x86_ops->get_nested_state ?
 			kvm_x86_ops->get_nested_state(NULL, NULL, 0) : 0;
 		break;
+	case KVM_CAP_HYPERV_ENLIGHTENED_VMCS:
+		r = kvm_x86_ops->nested_enable_evmcs != NULL;
+		break;
 	default:
 		break;
 	}
